realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -233,6 +234,7 @@ static RealmCacheType valueOf(Class<? extends BaseRealm> clazz) {
             this.realmClass = realmClass;
             this.callback = callback;
             this.notifier = notifier;
+            this.realmFileIsBeingCreated = !configuration.realmExists();
```

Comment:
How does this intervene with potential opens? I guess not very common use case, but shouldn't it somehow be synchronized with other calls to `getInstance` or is it always safe to execute the inner parts of `doCreateRealmOrGetFromCache` with `realmFileIsBeginCreated=true` even though the file has in fact been created from other side?

Reply:
This is only being used on `doCreateRealmOrGetFromCacheAsync`, so it should not intervene with other opens.